include/nbl/builtin/shader/loader/gltf/vertex_impl.glsl

```diff
@@ -0,0 +1,85 @@
+#include "nbl/builtin/shader/loader/gltf/common.glsl"
+
+#ifndef _NBL_VERT_INPUTS_DEFINED_
+#define _NBL_VERT_INPUTS_DEFINED_
+
+layout (location = _NBL_V_IN_POSITION_ATTRIBUTE_ID) in vec3 vPos;
+layout(location = _NBL_V_IN_NORMAL_ATTRIBUTE_ID) in vec3 vNormal;
+
+#ifndef _DISABLE_COLOR_ATTRIBUTES
+layout(location = _NBL_V_IN_COLOR_ATTRIBUTE_BEGINING_ID) in vec4 vColor_0;
+#ifdef _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_1_ENABLE
+#define _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_1 _NBL_V_IN_COLOR_ATTRIBUTE_BEGINING_ID + 1
+layout(location = _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_1) in vec4 vColor_1
+#endif // _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_1_ENABLE
+#ifdef _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_2_ENABLE
+#define _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_2 _NBL_V_IN_COLOR_ATTRIBUTE_BEGINING_ID + 2
+layout(location = _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_2) in vec4 vColor_2
+#endif // _NBL_EXTRA_V_IN_COLOR_ATTRIBUTE_ID_2_ENABLE
+#endif // _DISABLE_COLOR_ATTRIBUTES
+
+#ifndef _DISABLE_UV_ATTRIBUTES
+layout (location = _NBL_V_IN_UV_ATTRIBUTE_BEGINING_ID) in vec2 vUV_0;
+#ifdef _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_1_ENABLE
+#define _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_1 _NBL_V_IN_UV_ATTRIBUTE_BEGINING_ID + 1
+layout (location = _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_1) in vec2 vUV_1
+#endif // _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_1_ENABLE
+#ifdef _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_2_ENABLE
+#define _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_2 _NBL_V_IN_UV_ATTRIBUTE_BEGINING_ID + 2
+layout(location = _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_2) in vec2 vUV_2
+#endif // _NBL_EXTRA_V_IN_UV_ATTRIBUTE_ID_2_ENABLE
+#endif // _DISABLE_UV_ATTRIBUTES
+
+#if !defined(_DISABLE_COLOR_ATTRIBUTES) && !defined(_DISABLE_UV_ATTRIBUTES)
+#error "ERROR: UV and Color attributes must not be defined both either!"
+#endif
+
+#endif //_NBL_VERT_INPUTS_DEFINED_
+
+#ifndef _NBL_VERT_OUTPUTS_DEFINED_
+#define _NBL_VERT_OUTPUTS_DEFINED_
+
+/*
+    glTF Loader's shaders don't support multiple out attributes
+    at the moment, to change in future
+*/
+
+layout (location = _NBL_V_OUT_LOCAL_POSITION_ID) out vec3 LocalPos;
+layout (location = _NBL_V_OUT_VIEW_POS_ID) out vec3 ViewPos;
+#ifndef _DISABLE_UV_ATTRIBUTES
+layout (location = _NBL_V_OUT_UV_ID) out vec2 UV_0;
+#endif // _DISABLE_UV_ATTRIBUTES
+#ifndef _DISABLE_COLOR_ATTRIBUTES
+layout(location = _NBL_V_OUT_COLOR_ID) out vec4 Color_0;
```

Comment:
do you need the alpha?

ties into
https://github.com/Devsh-Graphics-Programming/Nabla/pull/9/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.glsl&file-filters%5B%5D=.h#r608812266